serrano/resources/field/base.py

```diff
@@ -82,7 +85,14 @@ def get_link_templates(self, request):
         }
 
     def get_queryset(self, request, params):
-        queryset = self.model.objects.all()
+        tree = trees[params['tree']]
+        q = Q()
+
+        # No public method for accessing the local models on the tree
+        for app_name, model_name in tree._models:
+            q |= Q(app_name=app_name, model_name=model_name)
+
+        queryset = self.model.objects.filter(q)
```

Comment:
I added an extra test case for a concept that contains two unrelated fields (this should never happen in practice) and it failed. Also, I noticed that the SQL being produced awful, excessive joins and what not. So I changed the concept filtering to use a subquery on the [unrelated fields](https://github.com/chop-dbhi/serrano/pull/258/files#diff-07ea7308a4ab10cdbb4073d4efa5b439R123) which is much more straightforward and the test passes.